crypto/ec/ec_ameth.c

```diff
@@ -513,8 +513,10 @@ int ec_pkey_export_to(const EVP_PKEY *from, void *to_keydata,
 
     if (pub_point != NULL) {
         /* convert pub_point to a octet string according to the SECG standard */
+        point_conversion_form_t format = EC_KEY_get_conv_form(eckey);
+
         if ((pub_key_buflen = EC_POINT_point2buf(ecg, pub_point,
-                                                 POINT_CONVERSION_COMPRESSED,
+                                                 format,
```

Comment:
I believe that the pyca failure is "caused" by this change.  In the (invalid) public key that they test against, the format number is 0 (which is invalid), and I suspect that it gets caught right about here:

https://github.com/openssl/openssl/blob/7489ada9f3fd902c5bc3c58cc03a90de2800d0ab/crypto/ec/ecp_oct.c#L169-L174

Before this, it would get the hard coded `POINT_CONVERSION_COMPRESSED` and happily create a point at infinity (the point value being `00`), which just fails later with the "point at infinity" error.

I consider the "point at infinity" error a bug, and the new behavior more correct.

Reply:
Adding the same dump of the key in question as I did in openssl/technical-policies#59:

``` console
$ openssl asn1parse -i -dump <<_____
-----BEGIN PUBLIC KEY-----
MBkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDAgAA
-----END PUBLIC KEY-----
_____
    0:d=0  hl=2 l=  25 cons: SEQUENCE          
    2:d=1  hl=2 l=  19 cons:  SEQUENCE          
    4:d=2  hl=2 l=   7 prim:   OBJECT            :id-ecPublicKey
   13:d=2  hl=2 l=   8 prim:   OBJECT            :prime256v1
   23:d=1  hl=2 l=   2 prim:  BIT STRING        
      0000 - 00 00                                             ..
```